lib/routes/dhu/jiaowu/news.ts

```diff
@@ -26,32 +28,60 @@ export const route: Route = {
     name: '教务处通知',
     maintainers: ['KiraKiseki'],
     handler,
-    description: `| 学生专栏 | 教师专栏 |
-  | -------- | -------- |
-  | student  | teacher  |`,
+    description: `| 学生专栏 | 教师专栏 | 选课专栏（仅选课期间开放） | 辅修专业 |
+  | -------- | -------- | -------- | -------- |
+  | student  | teacher  | class    | fxzy     |`,
 };
 
 async function handler(ctx) {
-    const type = ctx.req.param('type');
-    const link = Object.hasOwn(map, type) ? `${base_url}${map[type]}` : `${base_url}/tzggwxszl/list.htm`;
-    const response = await got({
-        method: 'get',
-        url: link,
-        headers: {
-            Referer: base_url,
-        },
-    });
-
-    const $ = load(response.data);
+    const type = ctx.req.param('type') || 'student';
+    const link = `${baseUrl}${map[type]}`;
+    const { data: response } = await got(link);
+
+    const $ = load(response);
+
+    // article list
+    const articleList = $('.list2 > li')
+        .toArray()
+        .map((item) => {
+            item = $(item);
+            const news_title = item.find('.news_title > a');
+            const news_meta = item.find('.news_meta');
+
+            const link = news_title.attr('href');
+            const title = news_title.text();
+            const pubDate = parseDate(news_meta.text(), 'YYYY-MM-DD', 'zh-cn');
+            return {
+                title,
+                link,
+                pubDate,
+                description: '',
+            };
+        });
+
+    // fetch article content
+    const items = await Promise.all(
+        articleList.map(async (item) => {
+            const url = `${baseUrl}${item.link}`;
```

Comment:
You can do this in
https://github.com/DIYgod/RSSHub/blob/1460a8447d840970d7791278301f1fe6738f5ddd/lib/routes/dhu/jiaowu/news.ts#L54-L59